crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -543,20 +553,40 @@ fn execute_syscall(
             Ok(())
         }
         DeprecatedSyscallSelector::Deploy => Err(HintError::CustomHint(Box::from(
-            "Use snforge_std::ContractClass::deploy instead of deploy_syscall"
-                .to_string(),
+            "Use snforge_std::ContractClass::deploy instead of deploy_syscall".to_string(),
         ))),
         DeprecatedSyscallSelector::ReplaceClass => Err(HintError::CustomHint(Box::from(
-            "Replace class can't be used in tests"
-                .to_string(),
+            "Replace class can't be used in tests".to_string(),
         ))),
-        DeprecatedSyscallSelector::GetBlockHash => Err(HintError::CustomHint(Box::from(
-            "Get block hash is temporarily disabled in tests: https://github.com/foundry-rs/starknet-foundry/issues/686"
-                .to_string(),
-        ))),        _ => blockifier_syscall_handler.execute_hint(vm, exec_scopes, hint_data, constants),
```

Comment:
Simply redirecting it doesn't work? Or we want to explicitly set the `BlockHash` to 0?

Reply:
Doesn't work since the storage is uninitialized (block hash is being stored on-chain)
We don't really have the concept of a block, and it always returns 0 in the contracts, so i'm just aligning it with the implementation here.

(I think it's related to [this one](https://github.com/foundry-rs/starknet-foundry/issues/732))
So when the above is fixed, we can just redirect, but the result will be the same still. 
To mock the hash, we can implement a cheatcode: https://github.com/foundry-rs/starknet-foundry/issues/721